src/general/ModLoader.cs

```diff
@@ -0,0 +1,320 @@
+        if (loadedItemList.GetSelectedItems().Length <= 0)
+        {
+            return;
+        }
+
+        var selectedItem = loadedItemList.GetSelectedItems()[0];
+        modList.Add(loadedModList[selectedItem]);
+        unloadedItemList.AddItem(loadedModList[selectedItem].ModName);
+        loadedModList.RemoveAt(selectedItem);
+        loadedItemList.RemoveItem(selectedItem);
+    }
+
+    private void OnRefreshPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        modList.Clear();
+        loadedModList.Clear();
+        unloadedItemList.Clear();
+        loadedItemList.Clear();
+        DirectoryInfo modFolder = Directory.CreateDirectory($"{Directory.GetCurrentDirectory()}\\mods");
+        foreach (DirectoryInfo currentMod in modFolder.EnumerateDirectories())
+        {
+            if (!File.Exists(currentMod.FullName + "/mod_info.json"))
+            {
+                continue;
+            }
+
+            var currentModInfo =
+                JsonConvert.DeserializeObject<ModInfo>(ReadJSONFile(currentMod.FullName + "/mod_info.json"));
+
+            if (currentModInfo.AutoLoad)
+            {
+                continue;
+            }
+
+            currentModInfo.Location = currentMod.FullName;
+            modList.Add(currentModInfo);
+            unloadedItemList.AddItem(currentModInfo.ModName);
+        }
+    }
+
+    private void OnLoadPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        if (loadedModList.Count <= 0)
+        {
+            return;
+        }
+
+        foreach (ModInfo currentMod in loadedModList)
+        {
+            if (string.IsNullOrEmpty(currentMod.Dll))
+            {
+                if (File.Exists(currentMod.Location + "/" + currentMod.Dll))
+                {
+                    Assembly.LoadFile(currentMod.Location + "/" + currentMod.Dll);
+                }
+            }
+
+            if (!File.Exists(currentMod.Location + "/mod.pck"))
+            {
+                GD.Print("Fail to find mod file: " + currentMod.ModName);
+                continue;
+            }
+
+            if (ProjectSettings.LoadResourcePack(currentMod.Location + "/mod.pck", true))
+            {
+                GD.Print("Loaded mod: " + currentMod.ModName);
+            }
+            else
+            {
+                GD.Print("Failed to load mod: " + currentMod.ModName);
+            }
+        }
+
+        GD.Print("All mods loaded");
+        SceneManager.Instance.ReturnToMenu();
+    }
+
+    private void OnBackPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        EmitSignal(nameof(OnModLoaderClosed));
+    }
+
+    private void OnResetPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        confirmationPopup.PopupCentered();
+    }
+
+    private void ResetGame()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        if (!File.Exists(Directory.GetCurrentDirectory() + "/Thrive.pck"))
```

Comment:
Would it be possible to ask godot what pck we have loaded? As the path might not always work. AFAIK you can rename the godot game executable and pck file and it'll still work. This would fail with that.

Reply:
It does not seem to be possible to do that but i could ask the player to select a file to reset from instead

Reply:
Another idea for reset, is to just require the game to be restarted. And before that marking all mods not to be loaded.

Reply:
I could do that